streamer.py

```diff
@@ -0,0 +1,97 @@
+from abc import ABC, abstractmethod
+from dotenv import load_dotenv
+import json
+import os
+
+load_dotenv()
+
+
+class RoleInterface(ABC):
+    @abstractmethod
+    def get_id(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_name(self) -> str:
+        pass
+
+
+class Role(RoleInterface):
+    def __init__(self, role_id: int, name: str):
+        self._id = role_id
+        self._name = name
+
+    def get_id(self) -> int:
+        return self._id
+
+    def get_name(self) -> str:
+        return self._name
+
+
+class StreamerInterface(ABC):
+    @abstractmethod
+    def get_id(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_username(self) -> str:
+        pass
+
+    @abstractmethod
+    def get_roles(self) -> list:
+        pass
+
+    @abstractmethod
+    def is_match(self, username) -> bool:
+        pass
+
+
+class Streamer(StreamerInterface):
+    def __init__(self, user_id: int, username: str, roles: list):
+        self._id = user_id
+        self._username = username
+        self._roles = roles
+
+    def get_id(self) -> int:
+        return self._id
+
+    def get_username(self) -> str:
+        return self._username
+
+    def get_roles(self) -> list:
+        return self._roles
+
+    def is_match(self, username) -> bool:
+        return self._username == username
+
+
+class MapperInterface(ABC):
+    @abstractmethod
+    def map(self) -> list:
+        pass
+
+
+class RoleMapper(MapperInterface):
+    def map(self, datasource: dict) -> list:
+        roles = []
+        for role in datasource:
+            roles.append(Role(role['role_id'], role['name']))
+
+        return roles
+
+
+class StreamerMapper(MapperInterface):
+    def map(self) -> list:
+        STREAMER_DATASOURCE = os.getenv('STREAMER_DATASOURCE')
```

Comment:
Todo: Change the StreamerMapper / RoleMapper to rely on the datasource (whitelist.py) classes rather than assuming it will be a json file. Adapt those classes to provide data to the mappers which they can turn into objects.